docs/aws-sts-provisioning.md

```diff
@@ -2,73 +2,55 @@
+`$PULL_SECRET` should be the path to your pull secret required for pulling the release image.
 
-Example:
-
-```yaml
-apiVersion: v1
-stringData:
-  credentials: |-
-    [default]
-    role_arn = arn:aws:iam::125931421481:role/mystsprefix-openshift-image-registry-installer-cloud-credentials
-    web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
-kind: Secret
-metadata:
-  name: installer-cloud-credentials
-  namespace: openshift-image-registry
-type: Opaque
+```
+CCO_IMAGE=$(oc adm release info --image-for='cloud-credential-operator' ${RELEASE_IMAGE}) && oc image extract ${CCO_IMAGE} --file='/usr/bin/ccoctl' --registry-config=${PULL_SECRET}
+chmod u+x ccoctl
 ```
 
+## Setup STS Infrastructure
 
-### Create Authentication Manifest
-
-In the same directory as your Secret manifests, add another file named `cluster-authentication-02-config.yaml` to configure the OpenShift Authentication operator to use the S3 OIDC provider created by `ccoctl create identity-provider`.
-
-```yaml
-apiVersion: config.openshift.io/v1
-kind: Authentication
-metadata:
-  name: cluster
-spec:
-  serviceAccountIssuer: https://mystsprefix-oidc.s3.us-east-1.amazonaws.com
+Create AWS resources using the [ccoctl](ccoctl.md#steps-create) tool (you will need aws credentials with sufficient permissions). The command below will generate public/private ServiceAccount signing keys, create the S3 bucket (with public read-only access), upload the OIDC config into the bucket, set up an IAM Identity Provider that trusts that bucket configuration, and create IAM Roles for each AWS CredentialsRequest extracted above. It will also dump the files needed by the installer in the `_output` directory. Installation secret manifests will be found within `_output/manifests`.
+```
+./ccoctl aws create-all --name <aws_infra_name> --region <aws_region> --credentials-requests-dir ./credrequests --output-dir _output/
 ```
 
-May also soon be automated by ccoctl.
+Hive allows providing arbitrary Kubernetes resource manifests to pass through to the install process. We will leverage this to inject the Secrets and configuration required for an STS cluster by creating a secret containing these manifests in the steps below. The manifests secret will be referenced from the ClusterDeployment.
 
 ## Create Hive ClusterDeployment
 
 Create a ClusterDeployment normally with the following changes:
 
-  1. Create a Secret for your private service account signing key created with ccoctl key-pair above: `kubectl create secret generic bound-service-account-signing-key --from-file=bound-service-account-signing-key.key=serviceaccount-signer.private`
-  1. Create a Secret for your installer manifests (credential role Secrets, Authentication config): `kubectl create secret generic cluster-manifests --from-file=manifests/`
+  1. Create a Secret for your private service account signing key created with `ccoctl aws create-all` above.
```

Comment:
This could be simplified in a FUP to utilize `hiveutil` for these steps,

```
hiveutil create-cluster sts-cluster \
--base-domain <basedomain> \
--ssh-private-key-file=<privkey> \
--ssh-public-key-file=<pubkey> \
--pull-secret-file=<pull-secret-path>\
--manifests=<path>/_output/manifests \
--bound-service-account-signing-key-file=<path>/output/serviceaccount-signer.private \
-o yaml > /tmp/sts-cluster.yaml

```

Reply:
Though we may wish to limit how much we suggest using hiveutil, as it's theoretically not a "supported" tool.